connectors/src/connectors/microsoft/temporal/workflows.ts

```diff
@@ -1,29 +1,98 @@
 import type { ModelId } from "@dust-tt/types";
-import { proxyActivities } from "@temporalio/workflow";
+import {
+  continueAsNew,
+  executeChild,
+  proxyActivities,
+  workflowInfo,
+} from "@temporalio/workflow";
 
 import type * as activities from "@connectors/connectors/microsoft/temporal/activities";
 import type * as sync_status from "@connectors/lib/sync_status";
-import type { DataSourceConfig } from "@connectors/types/data_source_config";
+import type { MicrosoftNodeType } from "@connectors/resources/microsoft_resource";
 
-const { fullSyncActivity } = proxyActivities<typeof activities>({
+const { getSiteNodesToSync, syncFiles, markNodeAsVisited } = proxyActivities<
+  typeof activities
+>({
   startToCloseTimeout: "20 minutes",
 });
 
-const { syncSucceeded } = proxyActivities<typeof sync_status>({
+const { reportInitialSyncProgress, syncSucceeded } = proxyActivities<
+  typeof sync_status
+>({
   startToCloseTimeout: "10 minutes",
 });
 
 export async function fullSyncWorkflow({
   connectorId,
-  dataSourceConfig,
+  startSyncTs = undefined,
 }: {
   connectorId: ModelId;
-  dataSourceConfig: DataSourceConfig;
+  startSyncTs?: number;
 }) {
-  await fullSyncActivity({ connectorId, dataSourceConfig });
+  if (startSyncTs === undefined) {
+    startSyncTs = new Date().getTime();
+  }
+
+  await executeChild(fullSyncSitesWorkflow, {
+    workflowId: microsoftFullSyncSitesWorkflowId(connectorId),
+    searchAttributes: {
+      connectorId: [connectorId],
+    },
+    args: [{ connectorId, startSyncTs }],
+    memo: workflowInfo().memo,
+  });
+}
+
+export async function fullSyncSitesWorkflow({
+  connectorId,
+  startSyncTs,
+  nodesToSync = undefined,
+  totalCount = 0,
+}: {
+  connectorId: ModelId;
+  startSyncTs: number;
+  nodesToSync?: MicrosoftNodeType[];
```

Comment:
We should avoid passing resources in workflow. Always use ids.

Also this name is confusing since in some places `MicrosoftNodeType` is imported from `@connectors/connectors/microsoft/lib/node_types`

Reply:
Similar to @spolu 's com, I take it you mean resources in the sense of "non-primitives types"? (not our resources, that I'm not passing here)

Ack, fixing

Reply:
Def agree the name is an issue, will also fix that